### ISSUE TITLE
Components set in Pretenders do not match the selector

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#1768]
- Markuplint version: `4.x.x`
- Parser lang: JSX
- Node.js version: `22.x.x`
- OS: macOS

## Describe the bug

Components set in Pretenders do not match the selector. The issue occurs with the following settings and code.

### Pretenders

```json
{
  "selector": "MyComponent",
  "as": "div"
}
```

### Component
```jsx
<MyComponent href="path/to" />
```

### nodeRules
```json
{
  "nodeRules": [
    {
      "selector": "MyComponent", // <= Didn't match
      "selector": "a", // <= Matched
      "rules": { "require-accessible-name": false }
    }
  ]
}
```

## The current behavior

`MyComponent` does not match the selector. 

## The expected behavior

The expected behavior is that both `a` and `MyComponent` should match.
